pyproject.toml

```diff
@@ -5,10 +5,13 @@ description = ""
 authors = ["GuillermoDR <70137647+GuillermoDR@users.noreply.github.com>"]
 
 [tool.poetry.dependencies]
-python = "^3.7"
-tldextract = "^3.1.0"
+python = "^3.11"
```

Comment:
@mattcristal @GuillermoDR I don't think we should use 3.11: this repo is used by other repos that do not use 3.11, so if I start editing this and use something like str | None instead of Optional[str], then the other repos will break

Reply:
also, the python requirement in the `setup.py` is much different https://github.com/dealroom/data-urlextract/blob/47ded503d832cf068f007fce3217df26f327f410/setup.py#L22